### ISSUE TITLE
Save should call validate and fail before saving

### ISSUE THREAD

**Issue by eshaiju** [amberframework/granite-orm#81]
I am very much new to Crystal and Amber. trying to create a basic blog application in amber.  I like to add presence validation in title and body of post model. how can we achieve this?

```
require "granite_orm/adapter/pg"

class Post < Granite::ORM::Base
  adapter pg

  field name : String
  field body : String
  field draft : Bool
  timestamps
end

```

**Comment by coderhs** [comment]
You can mention it inside the DB and access those errors when it fails to save. https://github.com/amberframework/granite-orm#errors

**Comment by eshaiju** [comment]
@coderhs I am asking about keywords equivalent to **  validates :name, presence: true**. I didn't find any in Doc

**Comment by coderhs** [comment]
So I just read through the code and I can see that its not properly documented, so thats a good candidate for a PR.

the validation can be added as so:

```rb
require "granite_orm/adapter/sqlite"

class Comment < Granite::ORM::Base
  adapter sqlite

  table_name post_comments
  field name : String
  field body : String

  validate(:name, "Should not be blank", ->(x : self) { x.name != nil && x.name != "" })
end
```

But the interesting point to note is that validation is not checked when you run `object.save`, only validations that are checked at that point are the database validation. If you want to check the validations you mention then you need to run a `object.valid?`.

When you run the `.valid?` method it will execute the block you passed in and if it failed then place the custom message.

PS: Even if the validators you placed fail it will still get saved to the DB, but if there are any DB level constrains that fail then it won't get saved. Unlike rails it won't raise an exception, but make those errors available in the `object.errors` array.

**Comment by Xosmond** [comment]
So there is not basic validations implemented ?

**Comment by drujensen** [comment]
@Xosmond This was fixed in #127
